### DISCUSSION TITLE
Upcoming change to worker startup procedure

### DISCUSSION THREAD

**Discussion by subpop** [RedHatInsights/yggdrasil#52] [General]
I'm planning to introduce a breaking change to the worker startup procedure in PR #37. A worker is no longer identified by the existence of an executable program in `/usr/libexec/yggdrasil/` with a filename ending in "`worker`". Workers will now only be identified by the presence of a configuration file in `/etc/yggdrasil/workers` that defines the location of the executable, among other properties.

**Comment by mkanoor** [comment]
@subpop When are these changes going to be released?

**Reply by subpop** [reply]
I don't know yet. My first goal is to get things merged into `main` so they will ship in the next `yggdrasil` release. But even that won't land in CentOS without my packaging it.

**Reply by subpop** [reply]
A little more info here; this will be merging to `main`, and will be part of the next "major" release of yggdrasil, version 0.3. Version 0.2.x will continue to start workers the implicit way. This allows us to be more deliberate about how long we decide to maintain the 0.2 branch. Existing distributions using 0.2 will continue to operate as expected without any disruptive change like this.

**Comment by subpop** [comment]
PR #37 has been merged to `main` and will be part of yggdrasil 0.3.
